### DISCUSSION TITLE
Decide future of no-log-password rule

### DISCUSSION THREAD

**Discussion by ssbarnea** [ansible/ansible-lint#1605] [Q&A]
As @felixfontein suggested several times the removal of the no-log-password rule due to its high level of false-positive and false-negative matches, I would like to make a survey among ansible-lint users regarding which direction we should take.

Please note that we already made a change on master that should make it only report when looping is used as for the other cases ansible build-in protection should be enough to avoid leaking secrets. Still, looping is known to easily cause undesired exposures.

[![](https://api.gh-polls.com/poll/01F7B3MYBF707HGZWB0PRWCNWC/keep%20rule%20as%20experimental%20to%20produce%20warnings)](https://api.gh-polls.com/poll/01F7B3MYBF707HGZWB0PRWCNWC/keep%20rule%20as%20experimental%20to%20produce%20warnings/vote)
[![](https://api.gh-polls.com/poll/01F7B3MYBF707HGZWB0PRWCNWC/make%20rule%20as%20opt-in)](https://api.gh-polls.com/poll/01F7B3MYBF707HGZWB0PRWCNWC/make%20rule%20as%20opt-in/vote)
[![](https://api.gh-polls.com/poll/01F7B3MYBF707HGZWB0PRWCNWC/remove%20rule)](https://api.gh-polls.com/poll/01F7B3MYBF707HGZWB0PRWCNWC/remove%20rule/vote)

**Comment by aminvakil** [comment]
More info on https://github.com/ansible-community/ansible-lint/discussions/1589.

**Reply by ssbarnea** [reply]
@decentral1se @geerlingguy @greg-hellings @dmsimard @sshnaidm I hope you will vote for one of the options. I want to see at least 10 votes on that.

**Reply by geerlingguy** [reply]
I think I've only seen the warning pop up once, so it's not a big deal to me either way.

**Comment by felixfontein** [comment]
The rule is now opt-in (#1620). I hope someone will rewrite it so it comes a bit closer to what it should actually do...
